api/api/docs/base_docs.py

```diff
@@ -77,3 +92,122 @@ def get_operation_id(self) -> str:
             else:
                 operation_tokens.append("detail")
         return "_".join(operation_tokens)
+
+
+source_404_message = "Invalid source 'name'. Valid sources are ..."
+source_404_response = OpenApiResponse(
+    NotFoundErrorSerializer,
+    examples=[
+        OpenApiExample(
+            name="404",
+            value={"detail": source_404_message},
+        )
+    ],
+)
+
+
+def build_source_path_parameter(media_type: MediaType):
+    valid_description = (
+        f"Valid values are source_names from the stats endpoint: "
+        f"https://api.openverse.engineering/v1/{media_type}/stats/."
+    )
+
+    return OpenApiParameter(
+        name="source",
+        type=str,
+        location=OpenApiParameter.PATH,
+        description=f"The source of {media_type}. {valid_description}",
+    )
+
+
+creator_path_parameter = OpenApiParameter(
+    name="creator",
+    type=str,
+    location=OpenApiParameter.PATH,
+    description="The name of the media creator. This parameter "
+    "is case-sensitive, and matches exactly.",
+)
+tag_path_parameter = OpenApiParameter(
+    name="tag",
+    type=str,
+    location=OpenApiParameter.PATH,
+    description="The tag of the media. Not case-sensitive, matches exactly.",
+)
+
+
+def get_collection_description(media_type, collection):
+    if collection == "tag":
+        return f"""
+Get a collection of {media_type} with a specific tag.
+
+This endpoint returns only the exact matches, case-insensitive matches for the
+specified tag. For example, 'birds' and 'birding' are not matches for 'bird'.
+To search within the tag values, or to match several tags, use the `search` endpoint
+with `tags` query parameter instead of `q` parameter. In this case, the matches will
+ not be exact, so 'cat' would match both 'cat' and 'cats'.
```

Comment:
Just a note that "case-insensitive" here is incorrect unless we normalise tag inputs (as mentioned in another one of my comments).

Regarding the clarification of exact/fuzzy matching, the last two sentences are technically documentation for the search endpoint, not this one. It's good information to have on the search endpoint, but is displaced here. I'm having a hard time coming up with a good alternative though, mostly because a full description of how matching
works for this endpoint is mutli-faceted:

- It matches entire tags, so multi-word tags must be multi-word inputs. An input of `dog` will not match works tagged `dog walking`. Nor would `runner` match `marathon runner`.
- It is whitespace sensitive, so `dog walking` is distinct from `dog walking `, though not sure how much that matters. I can't really think of a reasonable example for this that wouldn't just be obtuse. Is it worth documenting though? It _could_ be important, maybe for hyphenated terms, for example? Place names sometimes have hyphens in them and are sometimes left out for orthographic or other reasons. In production data we have both "Saint Pierre des Champs" and "Saint-Pierre-des-Champs" in tags.
- Diacritical marks are also important. "jalapeño" and "jalapeno" match different tags (should we "deburr" tags?)
- It doesn't do text analysis, so plural and conjugated forms of the same word are distinct ("dog"/"dogs")

The last one _feels_ most important to me, but the diacritical distinction could also be pretty important. Here's the only text I could come up with as an alternative, but is it too long? When I try to write a concise version, I end up basically re-writing what you've already got here. I'm struggling to decide what level of detail is _necessary_ for this documentation.

```suggestion
This endpoint matches a single tag, exactly and entirely.

Differences that will cause tags to not match are:
- upper and lower case letters
- diacritical marks
- hyphenation
- spacing
- multi-word tags where the query is only one of the words in the tag
- multi-word tags where the words are in a different order

Examples of tags that **do not** match:
- "Low-Quality" and "low-quality"
- "jalapeño" and "jalapeno"
- "Saint Pierre des Champs" and "Saint-Pierre-des-Champs"
- "dog walking" and "dog  walking" (where the latter has two spaces between the last two words, as in a typographical error)
- "runner" and "marathon runner"
- "exclaiming loudly" and "loudly exclaiming"

For non-exact or multi-tag matching, using the `search` endpoint's `tags` query parameter.
```

This makes me wonder whether there's actually a larger improvement to make here, where we have a separate piece of documentation describing what we mean by "exact" and "non-exact" matching and where, to be linked from here and other relevant endpoints. It would be a less technical version of https://docs.openverse.org/api/reference/search_algorithm.html#text-analysis-tokenization

Reply:
> The last one _feels_ most important to me, but the diacritical distinction could also be pretty important. Here's the only text I could come up with as an alternative, but is it too long? When I try to write a concise version, I end up basically re-writing what you've already got here. I'm struggling to decide what level of detail is _necessary_ for this documentation.

I think that for tags, it is more important to have the exact match. Even in case of plural vs singular: the users might want to filter out only the images where several items are present. For instance, `cats` implies that the image has more than one cat, which might be important for the user.
Since we already have a way to search (not match exactly) in tags using `tags=` parameter, I think it's important to add the exact match for the tags view.

As for the docs, I think extracting the description of what "exact" and "non-exact" means and linking to it from the shorter endpoint descriptions would be a good idea. @stacimc, what do you think?